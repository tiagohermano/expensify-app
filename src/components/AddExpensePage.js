import React from 'react';

const AddExpensePage = () => (
  <div>
    This is my Add Expense component
  </div>
);

export default AddExpensePage;
