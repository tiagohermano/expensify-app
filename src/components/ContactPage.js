import React from 'react';

const ContactPage = () => (
  <div>
    <h1>Contact</h1>
    <p>Contact me!</p>
  </div>
);

export default ContactPage;
