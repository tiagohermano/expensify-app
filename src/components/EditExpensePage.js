import React from 'react';

const EditExpensePage = () => (
  <div>
    This is my Edit Expense component
  </div>
);

export default EditExpensePage;
