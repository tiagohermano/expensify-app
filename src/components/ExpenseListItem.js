import React from 'react';

const ExpenseListItem = ({ description, amount, createdAt }) => (
  <div>
    <h3>{description}</h3>
    <p>{amount}</p>
    <p>{createdAt}</p>
  </div>
);

export default ExpenseListItem;
