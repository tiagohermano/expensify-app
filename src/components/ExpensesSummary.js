import React from 'react';
import { connect } from 'react-redux';
import { Link } from 'react-router-dom';
import totalExpenses from '../selectors/expenses-total';
import selectExpenses from '../selectors/expenses';
import numeral from 'numeral';

export const ExpensesSummary = (props) => {
  const expenseWord = props.expensesCount === 1 ? 'expense' : 'expenses';
  return (
    <div className='page-header'>
      <div className="content-container">
        <h1 className='page-header__title'>Viewing <span>{props.expensesCount}</span> {expenseWord} totalling <span>{numeral(props.expensesTotal/100).format('$0,0.00')}</span></h1>
        <div className='page-header__actions'>
          <Link className='button' to='/create'>Add Expense</Link>
        </div>
      </div>
    </div>
  )
};

const mapStateToProps = (state) => {
  const visibleExpenses = selectExpenses(state.expenses, state.filters);
  return {
    expensesTotal: totalExpenses(visibleExpenses),
    expensesCount: visibleExpenses.length
  }
};

export default connect(mapStateToProps)(ExpensesSummary);
