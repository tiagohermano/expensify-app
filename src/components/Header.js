import React from 'react';
import { NavLink } from 'react-router-dom';

const Header = () => (
  <div>
    <h1>Expensify</h1>
    <NavLink to="/" activeClassName="is-active" exact>Home</NavLink>
    <NavLink to="/create" activeClassName="is-active">Create Expense</NavLink>
    <NavLink to="/edit" activeClassName="is-active">Edit Expense</NavLink>
  </div>
);

export default Header;
