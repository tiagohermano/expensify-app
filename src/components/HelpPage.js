import React from 'react';

const HelpPage = () => (
  <div>
    This is my Help component
  </div>
);

export default HelpPage;
