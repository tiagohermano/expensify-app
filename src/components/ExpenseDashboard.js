import React from 'react';
import ExpenseList from './ExpenseList';
import ExpenseListFilters from './ExpenseListFilters';
import ExpensesSummary from './ExpensesSummary';

const ExpenseDashboard = () => (
  <div>
    <ExpenseListFilters />
    <ExpensesSummary />
    <ExpenseList />
  </div>
);

export default ExpenseDashboard;
