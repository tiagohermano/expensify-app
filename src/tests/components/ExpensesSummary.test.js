import React from 'react';
import { shallow } from 'enzyme';
import { ExpensesSummary } from '../../components/ExpensesSummary';
import expenses from '../fixtures/expenses';
import { addExpense } from '../../actions/expenses';

test('should correctly render Expenses Summary with 1 expense', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={1} expensesTotal={4563}/>);
  expect(wrapper).toMatchSnapshot();
});

test('should correctly render Expenses Summary with multiple expenses', () => {
  const wrapper = shallow(<ExpensesSummary expensesCount={9} expensesTotal={4563456}/>);
  expect(wrapper).toMatchSnapshot();
});
