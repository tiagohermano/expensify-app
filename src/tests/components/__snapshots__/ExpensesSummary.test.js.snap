// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`should correctly render Expenses Summary with 1 expense 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        1
      </span>
       
      expense
       totalling 
      <span>
        $45.63
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;

exports[`should correctly render Expenses Summary with multiple expenses 1`] = `
<div
  className="page-header"
>
  <div
    className="content-container"
  >
    <h1
      className="page-header__title"
    >
      Viewing 
      <span>
        9
      </span>
       
      expenses
       totalling 
      <span>
        $45,634.56
      </span>
    </h1>
    <div
      className="page-header__actions"
    >
      <Link
        className="button"
        replace={false}
        to="/create"
      >
        Add Expense
      </Link>
    </div>
  </div>
</div>
`;
